### PR TITLE
fix_code_generation

### PR DESCRIPTION
**Pull request by nitinmohan87**

### CODE REVIEW

File: security/http/codegen.go [hunk 1]
```diff
@@ -52,163 +73,221 @@ type SchemeData struct {
...
+				Schemes:              schemes,
+				ServiceName:          data.ServiceName,
+				MethodName:           data.Method.Name,
+			},
+			FuncMap: codegen.TemplateFuncs(),
+		})
+	}
+}
+
+// getSchemesForEndpoint collects the security schemes for the given endpoint.
```
Comment (raphael) [inline security/http/codegen.go]: style nit-pick: Go tends to favor concise names especially for private methods. How about `computeSchemes` instead of `getSchemesForEndpoint`?

File: security/http/codegen.go [hunk 2]
```diff
@@ -52,163 +73,221 @@ type SchemeData struct {
...
-				Data:    secureData,
-				FuncMap: codegen.TemplateFuncs(),
-			})
 		}
 	}
+	return schemes
+}
+
+// findMappedName returns the mapped name for the given attribute name.
+func findMappedName(p *goadesign.MappedAttributeExpr, keyAtt string) string {
```
Comment (raphael) [inline security/http/codegen.go]: I wonder if we shouldn't add that as a method of the `MappedAttributeExpr` struct instead? This is a legitimate case where one might not know whether the key exists a-priori which isn't a case that the struct was designed for originally. Maybe the struct could have a `FindKey` method that accepts the name of the key and returns the name of the corresponding element and `true` if there is one otherwise empty string and `false`?
Also the test for `obj` being `nil` is not needed as a mapped attribute type is always object (`NewMappedAttribute` panics otherwise).

Comment (nitinmohan87) [reply security/http/codegen.go]: Added the function like you suggested here - https://github.com/goadesign/goa/pull/1478/commits/d6a66808843b7eb4431cf9aa33e932cc993bf3a5

File: security/README.md [hunk 3]
```diff
@@ -17,6 +17,18 @@ override the defaults set at the API or service level.
 The code generated by goa makes the information needed to authorize available to
 the service methods.
 
+## Enabling the plugin
+
+Enable this plugin by importing its package with name `_` and its DSL package
```
Comment (raphael) [inline security/README.md]: Maybe something like:
------
To enable the plugin and make use of the security DSL simply import both the `security` and the `dsl` packages as follows:

```go
import (
   _ "goa.design/plugins/security"
   . "goa.design/plugins/security/dsl"
)
```
Note the use of the blank identifier to import the `security` package which is necessary as the package is imported solely for its side-effects (initialization).
-----
is a bit clearer?

File: security/design/aliases.go [hunk 4]
```diff
@@ -0,0 +1,170 @@
+//************************************************************************//
```
Comment (raphael) [inline security/design/aliases.go]: Can you remind me this is necessary? The example imports the goa `http` design package - not the security design package. Looking at the security plugin `design` package I don't see a reason it would have to be imported in user designs. I think this file is not necessary (and anything related to it in the Makefile if anything).

Comment (nitinmohan87) [reply security/design/aliases.go]: Doesn't the aliaser automatically create this file for us?

Comment (nitinmohan87) [reply security/design/aliases.go]: Deleted it!

**Comment by raphael** [comment]
Awesome PR, thank you!
